### PR TITLE
[CI visibility] System test - CI Visibility integration

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Description

Integrates system-test with CI Visibility.
We generate a JUnit files with custom dd_tags.
Github action workflow will call to datadog-ci script to upload generated JUnit reports

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [x] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [x] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -378,3 +385,193 @@ def pytest_sessionfinish(session, exitstatus):
...
+                    name=f"dd_tags[systest.case.release.library.{library_name}]",
+                    value=f"{test_release[library_name]}",
+                )
+        if outcome == "failed":
+            if bool(skip_reason):
+                tc_skipped = ET.SubElement(testcase, "skipped")
+                tc_skipped.text = skip_reason
+            else:
+                tc_failure = ET.SubElement(testcase, "failure")
+                tc_failure.text = "failed test"
```
Comment (robertomonteromiguel) [inline conftest.py:574]: TODO Get details of the failure

**Review by pawelchcki — APPROVED** [review]
LGTM. But I'm no Python expert @cbeauchesne - can you take a look as well ?

File: conftest.py [hunk 2]
```diff
@@ -378,3 +385,193 @@ def pytest_sessionfinish(session, exitstatus):
...
+
+    json_report_path = "logs/report.json"
+    junit_report_path = "logs/reportJunit.xml"
+
+    if not os.path.exists(json_report_path) or not os.path.exists(junit_report_path):
+        logger.warning("Not all required output reports found(report.json or reportJunit.xml)")
+        return
+
+    # Opening JSON file
+    f = open(json_report_path)
```
Comment (cbeauchesne) [inline conftest.py:400]:
```suggestion
    with open(json_report_path) as f:
        json.load(f)
```

File: conftest.py [hunk 3]
```diff
@@ -378,3 +385,193 @@ def pytest_sessionfinish(session, exitstatus):
...
+            test["skip_reason"],
+            test_doc,
+            test_rfc,
+            test_coverage,
+            test_release,
+        )
+
+    for testsuite in junit_report_root.findall("testsuite"):
+        # Test suite name will be the scanario name
+        testsuite.set("name", f"{os.environ.get('SYSTEMTESTS_SCENARIO','EMPTY_SCENARIO')}")
```
Comment (cbeauchesne) [inline conftest.py:451]:
```suggestion
        testsuite.set("name", os.environ.get('SYSTEMTESTS_SCENARIO','EMPTY_SCENARIO'))
```

File: conftest.py (hunk 1 above)
Comment (cbeauchesne) [reply conftest.py:574]: if you want to do so, you'll need to modify the code from 

```
data -> json -> xml
```

to 

```
get_data() -> json
get_data() -> xml
```

File: utils/scripts/update_dashboard_CI_visibility.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+if [ "$#" -ne 2 ] ; then
```
Comment (cbeauchesne) [inline utils/scripts/update_dashboard_CI_visibility.sh:1]: hashbang :)

File: utils/scripts/update_dashboard_CI_visibility.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+curl -X GET "https://api.datadoghq.com/api/v1/dashboard/${dashboard_id}" \
+-H "Accept: application/json" \
+-H "DD-API-KEY: ${DD_API_KEY}" \
+-H "DD-APPLICATION-KEY: ${DD_APP_KEY}")
+
+#Search in all template variables (no more than 4) trying to find DASHBOARD_VARIABLE_NAME pass as parameter
+#When I achieve to find the variable, I will change the default value with the last pipelineId + run attemp number 
+for counter in 0 1 2 3  
+do
+    dashboard_json_variable=$(jq -r ".template_variables[$counter].prefix"  <<< "$dashboard_json")
```
Comment (cbeauchesne) [inline utils/scripts/update_dashboard_CI_visibility.sh:23]: maybe worthy to have a look on jq `select` feature

File: utils/scripts/upload_results_CI_visibility.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
+if [ "$#" -ne 4 ] ; then
```
Comment (cbeauchesne) [inline utils/scripts/upload_results_CI_visibility.sh:1]: hashbang

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: utils/scripts/update_dashboard_CI_visibility.sh (hunk 5 above)
Comment (robertomonteromiguel) [reply utils/scripts/update_dashboard_CI_visibility.sh:25]: We will probably change the number of dashboard variables. This script is not final version

**Comment by cbeauchesne** [comment]
I converrt it as a draft, otherwise it pops in my task list :)

**Comment by cbeauchesne** [comment]
@Kyle-Verhoog parametric seems to be failing in a stable way, is there a way to mark the failing test as bug/flaky ?

**Comment by Kyle-Verhoog** [comment]
@cbeauchesne ah yeah looks like it's flaky, I'll look into this! We can just disable the test for now if it's being a pain

**Review by katiehockman — COMMENTED** [review]
@Kyle-Verhoog iiiinteresting. These were passing yesterday when I merged [this](https://github.com/DataDog/system-tests/pull/576). But I just pulled main and tested locally and it's failing. Taking a look..

**Comment by robertomonteromiguel** [comment]
> @Kyle-Verhoog iiiinteresting. These were passing yesterday when I merged [this](https://github.com/DataDog/system-tests/pull/576). But I just pulled main and tested locally and it's failing. Taking a look..

@katiehockman Could you add decorators to ignore this flaky tests?

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**